tests/unit/test_venv.py

```diff
@@ -799,6 +803,10 @@ def test_env_variables_added_to_pcall(tmpdir, mocksession, newconfig, monkeypatc
         assert env is not None
         assert "ENV_VAR" in env
         assert env["ENV_VAR"] == "value"
+        assert env["ESCAPED_VAR"] == "{value}"
+        assert env["ESCAPED_VAR2"] == r"\{value\}"
+        assert env["COLON"] == ";" if sys.platform == "win32" else ":"
+        assert env["TTY_VAR"] == "OFF_VALUE"
```

Comment:
these seem unrelated to the patch

Reply:
these are testing other parts of substitution internals which are not tested in this context.  If `export()` is to work correctly with a stable API, all of the substitution internal values need to be converted into externally understandable values.

For the record, using `{posargs}` in setenv is another case of special handling in the substitution system, and it fails strangely in setenv.

And another internal voodoo not tested is `file|` - I havent tried that yet.

Reply:
I've tested with `file|` - no problems here, but https://github.com/tox-dev/tox/issues/1694 raised for other issues.

I also retested with `{posargs}`, `{packages}` and `{opts}` and very odd behaviour raised in https://github.com/tox-dev/tox/issues/1695